libs/wingc/src/type_check.rs

```diff
@@ -28,6 +29,41 @@ use wingii::type_system::TypeSystem;
 use self::jsii_importer::JsiiImportSpec;
 use self::symbol_env::SymbolEnvIter;
 
+/// Visitor Pattern for listing the variables that are initialized in the class constructor.
+struct VisitClassInit<'a> {
+	fields: &'a mut Vec<String>,
+}
+
+impl<'a> VisitClassInit<'a> {
+	pub fn new(fields: &'a mut Vec<String>) -> Self {
+		Self { fields }
+	}
+}
+
+impl<'ast> Visit<'ast> for VisitClassInit<'ast> {
+	fn visit_scope(&mut self, node: &'ast Scope) {
+		for stmt in &node.statements {
+			visit::visit_stmt(self, stmt);
+		}
+	}
```

Comment:
I think you can remove this method and it will work the same, since this is the same as the method's default implementation [here](https://github.com/winglang/wing/blob/3e5534cc27d7d29f3d592025e19156a9984bcab6/libs/wingc/src/visit.rs#L78-L85)

Reply:
It works, I can't believe it works